### DISCUSSION TITLE
What's the future of styling layers in lonboard?

### DISCUSSION THREAD

**Discussion by zacdezgeo** [developmentseed/lonboard#578] [Ideas]
I wanted to start a discussion around layer styling in lonboard. As of now, the end-users are responsible for converting their variables into normalized values. A user could easily use typical step functions available in GIS software like Felt or QGIS, such as continuous steps, jenks, quantiles, standard deviation, geometric intervals, or even a custom function. However, would there be a benefit to having the functionality directly built into lonboard? Personally, I'd love to pass an attribute of my dataframe to define my layer style.

Similarly, I would like to know if there have already been discussions around other styling options, such as proportional points (changing the size of a point based on an attribute). I saw the [categorical colormap option](https://developmentseed.org/lonboard/latest/api/colormap/#lonboard.colormap.apply_continuous_cmap), but it doesn't seem to apply to sizing. 

There are some technical limitations here, but it would be interesting to understand what's possible. It might not make sense for lonboard to match styling feature parity with other GIS software, but is there any rule of thumb for when additional functionality should be supported?

**Comment by kylebarron** [comment]
I think ideally this would be done as an integration with mapclassify: https://github.com/developmentseed/lonboard/issues/136. My hunch is that it would be ideal to have some examples showing how it's done, rather than depending on mapclassify ourselves.

**Comment by zacdezgeo** [comment]
@kylebarron; should we define the different styling options we want to demo? I could potentially take a crack at that notebook example.

**Reply by kylebarron** [reply]
I think there's a couple different questions in your original comment. For classification algorithms like jenks, we should integrate with mapclassify. See also recent discussion in https://github.com/developmentseed/lonboard/issues/593.

You could extend categorical support for sizing, but right now the implementation of that helper function expects color input.
